public/app/core/logs_model.test.ts

```diff
@@ -700,6 +700,61 @@ describe('logSeriesToLogsModel', () => {
 
     expect(logSeriesToLogsModel(logSeries)).toMatchObject(metaData);
   });
+  it('should return correct metaData when some data frames have empty fields', () => {
+    const logSeries: DataFrame[] = [
+      toDataFrame({
+        fields: [
+          {
+            name: 'ts',
+            type: FieldType.time,
+            values: ['1970-01-01T00:00:01Z'],
+          },
+          {
+            name: 'line',
+            type: FieldType.string,
+            values: ['WARN boooo'],
+            labels: {
+              foo: 'bar',
+              baz: '1',
+              level: 'dbug',
+            },
+          },
+          {
+            name: 'id',
+            type: FieldType.string,
+            values: ['0'],
+          },
+        ],
+        refId: 'A',
+        meta: {
+          searchWords: ['test'],
+          limit: 1000,
+          stats: [{ displayName: 'Summary: total bytes processed', value: 97048, unit: 'decbytes' }],
+          custom: { lokiQueryStatKey: 'Summary: total bytes processed' },
+          preferredVisualisationType: 'logs',
+        },
+      }),
+      {
+        fields: [],
+        length: 0,
+        refId: 'B',
+        meta: {
+          searchWords: ['test'],
+          limit: 1000,
+          stats: [{ displayName: 'Summary: total bytes processed', value: 97048, unit: 'decbytes' }],
+          custom: { lokiQueryStatKey: 'Summary: total bytes processed' },
+          preferredVisualisationType: 'logs',
+        },
+      },
+    ];
+
+    const logsModel = dataFrameToLogsModel(logSeries, 0, 'utc');
+    expect(logsModel.meta).toMatchObject([
```

Comment:
Can we also check if we actually processed the lines from the dataframe that has fields?

Reply:
yes, done in a0b9ed1